src/BaseBinaryStar.cpp

```diff
@@ -1390,366 +1390,388 @@ void BaseBinaryStar::ResolveCoalescence() {
+ * - RTW 06/05/20
  * 
- * @param   [IN]    p_Resolve2ndSN              Indicates whether 2nd supernova should be evaluated regardless of bound/unbound status
+ * void EvaluateSupernovae()
  */
-void BaseBinaryStar::EvaluateSupernovae(const bool p_Resolve2ndSN) {
+void BaseBinaryStar::EvaluateSupernovae() { 
+
+    // Establish the SN State - which SNe occured and in what order
+    if (m_Star1->IsSNevent() && m_Star2->IsSNevent()) {																	// simultaneous supernovae
+		m_SupernovaState = SN_STATE::SIMUL;
+	} 
+	else if (m_Star1->IsSNevent()) {																					// only star1 SN at current timestep
 
-    m_SupernovaState = SN_STATE::NONE;                                                                                  // not yet determined
+        m_SupernovaState = (m_SupernovaState == SN_STATE::NONE) 									                    // star2 hasn't undergone supernova?
+                            ? SN_STATE::STAR10                                                                          // yes - just star1
+                            : SN_STATE::STAR21;                                                                         // no - star2 first, then star1
+	}
+	else if (m_Star2->IsSNevent()) {																					// only star2 SN at current timestep
 
-    double aPrime = m_SemiMajorAxisPrime;                                                                               // prior to processing SN events
+        m_SupernovaState = (m_SupernovaState == SN_STATE::NONE) 									                    // star1 hasn't undergone supernova?
+                            ? SN_STATE::STAR20                                                                          // yes - just star2
+                            : SN_STATE::STAR12;                                                                         // no - star1 first, then star2
+	}                                                                                                                   
 
-    if (m_Star1->IsSNevent() && (p_Resolve2ndSN || (utils::Compare(aPrime, 0.0) > 0))) {                                // star1 supernova
-        m_SupernovaState = SN_STATE::STAR1;                                                                             // star1
+    // Resolve the SN(e)
+    if (m_Star1->IsSNevent()) { 														                                // star1 supernova
 
         // resolve star1 supernova
         m_Supernova = m_Star1;                                                                                          // supernova
         m_Companion = m_Star2;                                                                                          // companion
-        (void)ResolveSupernova();                                                                                       // resolve supernova
+        (void)ResolveSupernova();                                                                               // resolve supernova
```

Comment:
@avigna , '(void)' here tells the compiler to ignore and discard any return value from the ResolveSupernova() function.  It's not strictly necessary (the compiler will ignore it anyway if you don't do something like 'x = ResolveSupernova()', but explicitly adding the '(void)' tells readers of the code that you didn't accidentally forget - you really do just want to ignore (and discard) the return value of the function (just so someone looking at the code doesn't think there might be a defect there).

Reply:
The void means ignore whatever is returned from the function. In this case, ResolveSupernova returns a boolean, but running the function by itself would leave a line which is just True/False and I think would throw an error. 

I'll leave it to @jeffriley to comment on why we need ResolveSupernova to be a boolean though. In principle, the function could return a void and I think it would all still work.

Reply:
It returns true if a SN event actually occurred, false if not (see the first IF statement in the function) - so the caller can check if anything actually changed during the call.  I think that's a hangover from the legacy code.